### PR TITLE
add pre-commit workflow to replace pre-commit-ci

### PR DESCRIPTION
**Pull request by braingram**
Since pre-commit-ci will no longer be auto-fixing PRs this PR also adds:
- a workflow that runs pre-commit (the branch protections will be updated when this PR is merged to make this a required job)
- a pull request template describing running pre-commit, pytest, etc

I also added an issue template to this PR since it seems like a good thing to provide.

Fixes https://github.com/asdf-format/asdf/issues/1679

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (WilliamJamieson) [inline .github/workflows/ci.yml:28]: See https://github.com/asdf-format/asdf/issues/1679#issuecomment-1810737318 as an alternative.

**Review by zacharyburnett — APPROVED** [review]
@braingram I still do not think the fact that outside collaborators can run CI without approval warrants disabling `pre-commit.ci` entirely, but I'll leave it to your prerogative as maintainer

**Pull request merged**
